HW18/HW18.py

```diff
@@ -0,0 +1,97 @@
+# 1
+class Bot:
+
+    def __init__(self, name):
+        self.name = name
+
+    def say_name(self):
+        print(self.name)
+
+    def send_message(self, message):
+        print(message)
+
+
+# 2
+class TelegramBot(Bot):
+
+    def __init__(self, name, url=None, chat_id=None):
+        self.url = url
+        self.chat_id = chat_id
+        super().__init__(name)
+
+    def send_message(self, message):
+        print(f'{self.name} bot says {message} to chat {self.chat_id} using {self.url}')
+
+    def set_chat_id(self, chat_id1):
+        self.chat_id = chat_id1
+
+    def set_url(self, url1):
+        self.url = url1
+
+
+# 3
+class MyStr(str):
```

Comment:
You can avoid of creation of a constructor here by using `self.upper()` in the `__str__`